spec/support/pdf_matchers.rb

```diff
@@ -3,30 +3,56 @@
 require "rspec/expectations"
 require "pdf-reader"
 
+module Helpers
+  module PdfMatchersHelper
+    def self.extract_pdf_content(pdf_stream)
+      temp_pdf = Tempfile.new("pdf")
+
+      ## Most linux systems distinguish between binary and text content files.
+      ## The PDF content is not compatible with file managed in text mode.
+      ## Tempfile#binmode will tell the system to manage the file in binmode rather
+      ## than text mode. This issue does not occur on OSX but it does in most
+      ## production servers.
+      temp_pdf.binmode
+
+      temp_pdf << pdf_stream
+      pdf_text = PDF::Reader.new(temp_pdf).pages.map(&:text).join
+      pdf_text = clean_pdf_text(pdf_text)
+      temp_pdf.close
+
+      pdf_text
+    end
+
+    def self.clean_pdf_text(text)
+      text.delete("\n").delete(" ").delete("\t")
+    end
+  end
+end
+
 RSpec::Matchers.define :match_pdf_content do |fixtures_path_to_expected_pdf|
   match do |actual|
     expected = PDF::Reader.new(fixtures_path_to_expected_pdf)
     @expected = expected.pages.map(&:text).join
-    @expected = @expected.delete("\n").delete(" ").delete("\t")
+    @expected = Helpers::PdfMatchersHelper.clean_pdf_text(@expected)
 
-    temp_pdf = Tempfile.new("pdf")
+    @actual = Helpers::PdfMatchersHelper.extract_pdf_content(actual)
+    @actual == @expected
+  end
 
-    ## Most linux systems distinguish between binary and text content files.
-    ## The PDF content is not compatible with file managed in text mode.
-    ## Tempfile#binmode will tell the system to manage the file in binmode rather
-    ## than text mode. This issue does not occur on OSX but it does in most
-    ## production servers.
-    temp_pdf.binmode
+  failure_message do |_actual|
+    "expected that \n#{@actual} \n would be equal to \n#{@expected}"
+  end
+end
 
-    temp_pdf << actual
-    @actual = PDF::Reader.new(temp_pdf).pages.map(&:text).join
-    @actual = @actual.delete("\n").delete(" ").delete("\t")
-    temp_pdf.close
+RSpec::Matchers.define :include_pdf_content do |expected|
```

Comment:
@Cruikshanks I was thinking that we should probably get rid of the `match_pdf_content` and only use this instead. Will make it clearer what information has actually been rendered in the PDF when reading the test. Here is the proposal code: https://github.com/DEFRA/waste-exemptions-engine/compare/223-tris...223-tris-matchers-proposal?expand=1 

Benefits: 
 - No need for Timecop, information can stay dynamic 
 - Allow the usage of gems like [Faker](https://github.com/stympy/faker/) that can really help to dry out the Factories 
 
The only drawback I can see is the length of the test. 
But, I actually prefer to be able to open a test and understand all that is going on. Error messages and regression fixes might also come much easier as the error will tell us which one is the information that has changed.